uniffi_core/src/ffi_converter_traits.rs

```diff
@@ -253,3 +263,16 @@ where
 
     const TYPE_ID_META: MetadataBuffer = T::TYPE_ID_META;
 }
+
+/// Trait used to lift references
+///
+/// This is needed because if we see a `&T` parameter, it's not clear which FfiConverter to use to
+/// lift it.  Normally it's just `T`, but for interfaces it's `Arc<T>` and for callback interfaces
+/// it's `Box<T>`.
+///
+/// This trait provides the proc-macros with the functionality they need to handle this
+pub trait LiftRef<UT> {
+    type LiftType: FfiConverter<UT>;
+
+    fn as_ref(v: &Self::LiftType) -> &Self;
+}
```

Comment:
I'm hoping `LiftRef` is the first step towards deconstructing `FfiConverter` into smaller traits, maybe those smaller traits could be named `Lift`, `Lower`, and `Return`.

I tried to avoid having to define the `as_ref` method.  I thought I could use the standard `Deref` or `AsRef`, but I couldn't get them to work for both `&T` and `&Arc<T>`, since the latter needs `&**` to convert it to the proper reference type.  If anyone else can get this to work, we can drop this method and the trait gets very simple.

Reply:
I think the `Borrow` trait could work here.

Reply:
That's beautiful, thanks.